### ISSUE TITLE
Password is not required for user creation as an admin

### ISSUE THREAD

**Issue by basilelegal** [tracim/tracim#1127]
**Describe the bug**

Even if emails are disabled you can create a user without a password. Neither Frontend or Backend shows anything. Only "User created" message appears.

**To Reproduce**
1. Go to `/ui/admin/user`
2. Click on `create user`
3. Enter a `username`,` email`, `role` and click on submit button.

**Expected behavior**
Backend should return an error saying that emails are not enabled.
We could add a checkbox in the Frontend  for asking to generate a token so the admin can send it to the user.

**Comment by PhilippeAccorsi** [comment]
@Skylsmoi The flash message says if you tried to create user without all information : add name, email, password and profile.

- [ ] verify if password is writed before create user

- [ ] check password lenght

@inkhey for backend idea, we need to discuss with @lebouquetin for an another version of tracim.

**Comment by PhilippeAccorsi** [comment]
- [ ] ] if the four necessary thing is not enter, dont activate create user button

**Comment by Skylsmoi** [comment]
this is **only** when emails are disabled

**Comment by PhilippeAccorsi** [comment]
Verified about frontend.

I move this issue in tracim V2.2 for backend.

**Comment by PhilippeAccorsi** [comment]
@inkhey make sure its not possible to create user without password in backend.
 This is for case when email is not activated.

**Comment by lebouquetin** [comment]
@inkhey let's verify if there is a check for that case in API, then add the check if not yet implemented + add associated test case (no notifs + no password = HTTP 400)

**Comment by inkhey** [comment]
@lebouquetin, @PhilippeAccorsi 
Current behavior of backend is this :
- if we send password -> password need more than x characters, else return 400
- if we do not send password -> 
   - if email notification enabled -> generate password for user.
   - if email notificated disabled -> create user without password (cannot logging)

It may seems logic to not accept user without password but : 
- How should we handle future mecanism where instead of returning first password we do return token for user to create it's first password : during this period user doesn't have a password. (or should we handle it differently ?)
- How should we handle other auth case, it doesn't make sense to add password for ldap user for example.

**Comment by PhilippeAccorsi** [comment]
~~~
How should we handle future mecanism where instead of returning first password we do return token for user to create it's first password : during this period user doesn't have a password. (or should we handle it differently ?)
~~~
@inkhey this case not exist because email notification is deactivated.

**Comment by PhilippeAccorsi** [comment]
~~~
How should we handle other auth case, it doesn't make sense to add password for ldap user for example.
~~~
In this case we can accept to create user without password if authentication type is external. I think.

**Comment by inkhey** [comment]
> `How should we handle other auth case, it doesn't make sense to add password for ldap user for example.`
> 
> In this case I think we can accept to create user without password if authentication type is external.

But currently we does not provide information about auth mecanism when creating account, we create an account with auth type is unknown by default and them at first authentification auth type become fixed.

**Comment by inkhey** [comment]
> `ere instead of returning first password we do return token for user to create it's first password : during this period user doesn't have a password. (or should we handle it differently ?)`

Yes, actually this case doesn't exist, what i mean is "when should i refuse account creation without password".
There is few case where is does means sense to have such account like i explain.

**Comment by Sarah-Pelle** [comment]
Fixed
